src/plugins/foxPage/components/FoxyAssetMarketData.tsx

```diff
@@ -0,0 +1,41 @@
+import { AssetId } from '@shapeshiftoss/caip'
+import { foxyAddresses } from '@shapeshiftoss/investor-foxy'
+import { useFoxy } from 'features/defi/contexts/FoxyProvider/FoxyProvider'
+import { useCallback, useEffect, useState } from 'react'
+import { AssetMarketData } from 'components/AssetHeader/AssetMarketData'
+import { selectAssetById } from 'state/slices/selectors'
+import { selectMarketDataById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type FoxyAssetMarketDataProps = {
+  assetId: AssetId
+}
+
+export const FoxyAssetMarketData = ({ assetId }: FoxyAssetMarketDataProps) => {
```

Comment:
Not sure If I should directly integrate this specific FOXy logic within `AssetMarketData` component ?

Reply:
Definitely not, anything that's not generic should augment, not shoehorn into the original base component IMO

Reply:
With that being said, can't this be a hook vs. a full component, if all we need is some additional data?

Reply:
Extracted to `useFoxyMarketData` hook